### DISCUSSION TITLE
Upload new photos results in duplicates.

### DISCUSSION THREAD

**Discussion by tinohager** [LycheeOrg/Lychee#2611] [Bugs] [answered]
### Lychee version

v6.0.0

### Which PHP version are you using?

PHP 8.2

### Detailed description of the problem

After uploading new photos, each photo was created 3 times in the album. Instead of 36 photos, I now have 108 photos in the album. And always 3x the same. On a test with 2 photos i have after upload 12 photos.

### Steps to reproduce the issue

1. Upload new Photos in a Album

### Diagnostics [REQUIRED]

Errors
warning
zend.assertions is disabled although Lychee is in debug mode. For easier debugging code generation for assertions should be enabled.
warning
APP_URL is still set to default, this will break access to all your images
info
Latest version of PHP is 8.3
warning
Default timezone not properly set; you might experience strange results when importing photos without explicit EXIF timezone
info
git (software) is not available.
error
Photo without Original found -- IMG_4145.JPG in Test1
error
Photo without Original found -- IMG_4151.JPG in Test1
info
Found 22 small that could be generated.
info
Found 23 small2x that could be generated.
info
Found 27 medium that could be generated.
info
Found 29 medium2x that could be generated.
info
Have you considered supporting Lychee? :)
Info
Lychee Version (tag):                      v6.0.0 (d25cd64) -- Up to date (-6.3926940639269E-11 years ago).
DB Version:                                6.0.0
composer install:                          --no-dev
APP_ENV:                                   local
APP_DEBUG:                                 true
APP_URL:                                   default
APP_DIR:                                   default
LOG_VIEWER_ENABLED:                        true
VUEJS_ENABLED:                             true
PHOTO_PIPES:                               true
System:                                    Linux
PHP Version:                               8.2.24
PHP User agent:                            Lychee/4 (https://lycheeorg.github.io/)
Timezone:                                  UTC
Max uploaded file size:                    100M
Max post size:                             100M
Chunk size:                                25.60 MB
Max execution time:                        3600
MySQL Version:                             10.11.9-MariaDB-ubu2204
exec() Available:                          yes
Imagick Available:                         1
Imagick Enabled:                           1
Imagick Version:                           1691
GD Version:                                2.3.3
Number of foreign key:                     12 found.

### Browser & System [REQUIRED]

chrome, docker:latest

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by tinohager** [comment]
Upload one image create three POST requests
![image](https://github.com/user-attachments/assets/b09c316f-739d-4960-8822-545aaf686af9)

**Comment by d7415** [comment]
What method are you using to upload?

**Comment by tinohager** [comment]
File Upload via the Webinterface. Click on the Drap and Drop area and choose some photos in the File System.

**Comment by tinohager** [comment]
![image](https://github.com/user-attachments/assets/44b0a0ca-5fe0-416a-9568-ab73505e09d0)
![image](https://github.com/user-attachments/assets/01e247be-3fbf-467d-9aaf-4f5bc0e3107a)
Upload 7 Images
![image](https://github.com/user-attachments/assets/8ffe43d4-bebb-471f-961e-f01c989c137a)
![image](https://github.com/user-attachments/assets/2065de41-1153-48dc-9ecc-fb2eed9d1222)
Now we have 14 Images started from an empty album
![image](https://github.com/user-attachments/assets/15630194-9d77-4f10-8c87-46a990ac3a4d)

**Reply by ildyria** [reply]
I am not able to reproduce on Linux, maybe a windows thing?

**Reply by tinohager** [reply]
![image](https://github.com/user-attachments/assets/35b66b85-4ec6-49e6-bafd-832b4fa29bb8)
Several upload processes are created immediately after selection. With the same file, it doesn't happen every time. I use the Docker container and have completely reset the entire setup. I run all local on my dev machine.

**Comment by tinohager** [comment]
The more often I create a new album, the more often the same photo is uploaded several times. Interestingly, a browser refresh does not completely defuse the issue.

**Comment by Borisvl** [comment]
I have the same issue using Safari on MacOs with a Linux server v6.0.1. 
Enabling duplicate detection can be used as a workaround. Then the upload window shows failed uploads, but at least they do not appear multiple times in the album.
<img width="777" alt="image" src="https://github.com/user-attachments/assets/d44cb188-efa3-4b09-b39a-ed4b0eab38f2">

**Comment by ferenckumi** [comment]
I have the same issue on Linux server with Ubuntu 24.04 LTS  apache2, PHP 8.3 and mysql Lychee V6.0.1
i have every picture 2times 3times now in the neu albums , just ask for more details.. when needed

**Reply by jessienab** [reply]
Also happening to me using the lychee Docker image running under Podman, v6.0.1, PHP 8.2, Arch Linux as host.

**Comment by bilbaomtl** [comment]
Same issue 
On my android phone  and on my ubuntu studio machine with chronium, on firefox and chrome.
there's my installation data:
Errors
info
Have you considered supporting Lychee? :)
Info
Lychee Version (git):                      master (8137557) -- 4 commits behind 42b4c20 (-8.4096875E-5 years ago)
DB Version:                                6.0.1
composer install:                          --no-dev
APP_ENV:                                   production
APP_DEBUG:                                 false
APP_URL:                                   set
APP_DIR:                                   default
LOG_VIEWER_ENABLED:                        true
VUEJS_ENABLED:                             true
PHOTO_PIPES:                               true
System:                                    Linux
PHP Version:                               8.3.13
PHP User agent:                            Lychee/4 (https://lycheeorg.github.io/)
Timezone:                                  America/Toronto
Max uploaded file size:                    500M
Max post size:                             500M
Chunk size:                                12.80 MB
Max execution time:                        200
MySQL Version:                             10.11.6-MariaDB-0+deb12u1
exec() Available:                          yes
Imagick Available:                         1
Imagick Enabled:                           1
Imagick Version:                           1691
GD Version:                                2.3.3
Number of foreign key:                     12 found.

		
Space
Configuration
version:                                   060001
check_for_updates:                         0
sorting_photos_col:                        taken_at
sorting_photos_order:                      ASC
sorting_albums_col:                        max_taken_at
sorting_albums_order:                      ASC
imagick:                                   1
skip_duplicates:                           1
small_max_width:                           0
small_max_height:                          360
medium_max_width:                          1920
medium_max_height:                         1080
lang:                                      fr
image_overlay_type:                        desc
default_license:                           none
compression_quality:                       90
grants_full_photo_access:                  1
delete_imported:                           0
mod_frame_enabled:                         1
mod_frame_refresh:                         30
thumb_2x:                                  1
small_2x:                                  1
medium_2x:                                 1
landing_page_enable:                       0
force_migration_in_production:             0
search_public:                             0
hide_version_number:                       0
SL_enable:                                 0
SL_for_admin:                              0
recent_age:                                1
grants_download:                           0
photos_wraparound:                         1
map_display:                               0
zip64:                                     1
map_display_public:                        0
map_provider:                              Wikimedia
force_32bit_ids:                           0
map_include_subalbums:                     0
update_check_every_days:                   3
has_exiftool:                              0
share_button_visible:                      0
import_via_symlink:                        0
has_ffmpeg:                                0
location_decoding:                         0
location_decoding_timeout:                 30
location_show:                             1
location_show_public:                      0
rss_enable:                                0
rss_recent_days:                           7
rss_max_items:                             100
prefer_available_xmp_metadata:             0
editor_enabled:                            1
lossless_optimization:                     0
swipe_tolerance_x:                         150
swipe_tolerance_y:                         250
local_takestamp_video_formats:             .avi|.mov
log_max_num_line:                          1000
unlock_password_photos_with_url_param:     0
nsfw_visible:                              1
nsfw_blur:                                 0
nsfw_warning:                              0
nsfw_warning_admin:                        0
nsfw_banner_override:                      
map_display_direction:                     1
album_subtitle_type:                       oldstyle
upload_processing_limit:                   4
new_photos_notification:                   0
legacy_id_redirection:                     1
zip_deflate_level:                         6
default_album_protection:                  1
album_decoration:                          layers
album_decoration_orientation:              row
allow_username_change:                     1
auto_fix_orientation:                      1
use_job_queues:                            0
random_album_id:                           starred
use_last_modified_date_when_no_exif_date:  0
layout:                                    justified
date_format_photo_thumb:                   M j, Y, g:i:s A e
date_format_photo_overlay:                 M j, Y, g:i:s A e
date_format_sidebar_uploaded:              M j, Y, g:i:s A e
date_format_sidebar_taken_at:              M j, Y, g:i:s A e
date_format_hero_min_max:                  F Y
date_format_hero_created_at:               M j, Y, g:i:s A T
date_format_album_thumb:                   M Y
upload_chunk_size:                         0
nsfw_banner_blur_backdrop:                 0
search_pagination_limit:                   1000
search_minimum_length_required:            4
photo_layout_justified_row_height:         320
photo_layout_masonry_column_width:         300
photo_layout_grid_column_width:            250
photo_layout_square_column_width:          200
photo_layout_gap:                          12
display_thumb_album_overlay:               always
display_thumb_photo_overlay:               hover
default_album_thumb_aspect_ratio:          1/1
use_album_compact_header:                  0
login_button_position:                     left
back_button_enabled:                       0
back_button_text:                          Return to Home
current_job_processing_visible:            1
SA_random_thumbs:                          0
job_processing_queue_visible:              1
login_required:                            0
enable_unsorted:                           1
enable_starred:                            1
enable_recent:                             1
enable_on_this_day:                        1
login_required_root_only:                  1
dark_mode_enabled:                         1
show_keybinding_help_popup:                0
show_keybinding_help_button:               1
disable_se_call_for_actions:               0
enable_se_preview:                         0
slideshow_timeout:                         5
keep_original_untouched:                   1
hide_nsfw_in_smart_albums_and_search:      1

**Comment by tinohager** [comment]
@ildyria 

The bug with the multiple uploads is caused because the `UploadPanel` is loaded twice.

- resources\js\components\headers\AlbumsHeader.vue
- resources\js\components\headers\AlbumHeader.vue

There are then 2 components that process upload orders from the store

**Reply by ildyria** [reply]
Yes but with `v-if` should remove/disable it.
Further more it does not explain why it would upload 3 / 4 / 5 times as you showed me the other day.

**Reply by tinohager** [reply]
I removed it from the code in my setup as a test and then had an upload

**Reply by tinohager** [reply]
`v-if` is only used if the user does not have upload authorization

**Reply by tinohager** [reply]
i have add an unmounted and mounted event and the unmount is not triggered in `AlbumsHeader.vue`

**Comment by jeanas** [comment]
I have the same issue. Let me know if I can do anything to help maintainers debug this.

**Reply by jeanas** [reply]
It may be worthy of note that I'm also seeing JS errors like

```
TypeError: a.value is undefined
    n https://selim.abou-samra.fr/lychee/build/assets/app-DQspYxnW.js:6
    Os https://selim.abou-samra.fr/lychee/build/assets/@vue-DA35nvq2.js:13
    st https://selim.abou-samra.fr/lychee/build/assets/@vue-DA35nvq2.js:13
    s https://selim.abou-samra.fr/lychee/build/assets/@vue-DA35nvq2.js:18
[@vue-DA35nvq2.js:13:2006](https://selim.abou-samra.fr/lychee/build/assets/@vue-DA35nvq2.js)
    Uu https://selim.abou-samra.fr/lychee/build/assets/@vue-DA35nvq2.js:13
    ls https://selim.abou-samra.fr/lychee/build/assets/@vue-DA35nvq2.js:13
    Os https://selim.abou-samra.fr/lychee/build/assets/@vue-DA35nvq2.js:13
    st https://selim.abou-samra.fr/lychee/build/assets/@vue-DA35nvq2.js:13
    s https://selim.abou-samra.fr/lychee/build/assets/@vue-DA35nvq2.js:18
```

**Comment by tinohager** [comment, marked as answer]
i tested the last dev docker container looks pretty good now
`docker pull lycheeorg/lychee:dev`

**Reply by tinohager** [reply]
@ildyria When can we expect the next release?

**Reply by ildyria** [reply]
When 2681 is reviewed and merged. 😅
